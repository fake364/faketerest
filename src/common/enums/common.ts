export enum GENDER_ENUM {
  FEMALE = -1,
  NOT_SPECIFIED = 0,
  MALE
}
