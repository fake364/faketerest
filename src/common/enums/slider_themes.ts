export enum SLIDER_THEMES {
	DINNER_IDEA = 'dinnerIdea',
	HOME_DECOR = 'homeDecor',
	NEW_OUTFIT = 'newOutfit',
	THUMB_IDEA = 'thumbIdea'
}
