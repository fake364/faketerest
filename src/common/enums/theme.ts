export enum THEME_TYPE {
	BASE = 'base',
	BUSINESS = 'business',
	NEWSROOM = 'newsroom'
}