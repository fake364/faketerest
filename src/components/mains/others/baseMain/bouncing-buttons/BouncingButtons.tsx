// eslint-disable-next-line max-len
import RoundedArrowButton from '../../../../../common/components/buttons/icon-button/prepared-components/RoundedArrowButton';
import React, { FC } from 'react';
import { SLIDER_THEMES } from '../../../../../common/enums/slider_themes';

type Props = {
  isAnimated: boolean;
  shownType: SLIDER_THEMES;
  onClickUp: () => void;
  onClickDown: () => void;
};
export const BouncingButtons: FC<Props> = ({
  isAnimated,
  shownType,
  onClickUp,
  onClickDown
}) => (
  <>
    {isAnimated && (
      <RoundedArrowButton
        shownType={shownType}
        additionalStyles={'bouncing-arrow-button arrow-button-up'}
        onClick={onClickDown}
      />
    )}
    {!isAnimated && (
      <RoundedArrowButton
        shownType={shownType}
        additionalStyles={'bouncing-arrow-button arrow-button-down mt-[24px]'}
        isArrowUp
        onClick={onClickUp}
      />
    )}
  </>
);
