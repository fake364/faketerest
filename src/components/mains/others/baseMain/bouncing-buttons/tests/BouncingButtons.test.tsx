import React from 'react';
import { shallow } from 'enzyme';
import { BouncingButtons } from '../BouncingButtons';
import { SLIDER_THEMES } from '../../../../../../common/enums/slider_themes';

describe('BouncingButtons', () => {
  it('should render animated down button', () => {
    const wrapper = shallow(
      <BouncingButtons
        isAnimated={true}
        shownType={SLIDER_THEMES.HOME_DECOR}
        onClickUp={jest.fn()}
        onClickDown={jest.fn()}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render not animated up button', () => {
    const wrapper = shallow(
      <BouncingButtons
        isAnimated={false}
        shownType={SLIDER_THEMES.HOME_DECOR}
        onClickUp={jest.fn()}
        onClickDown={jest.fn()}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
