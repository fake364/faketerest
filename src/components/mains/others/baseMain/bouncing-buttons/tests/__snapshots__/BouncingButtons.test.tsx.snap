// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BouncingButtons should render animated down button 1`] = `
<Fragment>
  <RoundedArrowButton
    additionalStyles="bouncing-arrow-button arrow-button-up"
    onClick={[MockFunction]}
    shownType="homeDecor"
  />
</Fragment>
`;

exports[`BouncingButtons should render not animated up button 1`] = `
<Fragment>
  <RoundedArrowButton
    additionalStyles="bouncing-arrow-button arrow-button-down mt-[24px]"
    isArrowUp={true}
    onClick={[MockFunction]}
    shownType="homeDecor"
  />
</Fragment>
`;
